src/uct/cuda/cuda_copy/cuda_copy_md.c

```diff
@@ -245,14 +251,17 @@ uct_cuda_copy_mem_alloc_fabric(uct_cuda_copy_alloc_handle_t *alloc_handle)
     prop.location.type        = CU_MEM_LOCATION_TYPE_DEVICE;
     prop.location.id          = cu_device;
 
-    status = UCT_CUDADRV_FUNC_LOG_ERR(cuMemGetAllocationGranularity(&granularity, &prop,
-                                          CU_MEM_ALLOC_GRANULARITY_MINIMUM));
-    if (status != UCS_OK) {
-        return status;
+    if (md->granularity == -1) {
```

Comment:
1. pls use SIZE_MAX instead of -1
2. cache also whether fabric allocation is supported